kotlinpoet-elementhandler-reflective/src/main/kotlin/com/squareup/kotlinpoet/elementhandler/reflective/ReflectiveElementHandler.kt

```diff
@@ -186,6 +186,27 @@ class ReflectiveElementHandler private constructor() : ElementHandler {
     }
   }
 
+  override fun parameterAnnotations(
+    classJvmName: String,
+    methodSignature: JvmMethodSignature,
+    index: Int,
+    isConstructor: Boolean
+  ): List<AnnotationSpec> {
+    val parameters = if (isConstructor) {
+      lookupConstructor(classJvmName, methodSignature)!!.parameters
+    } else {
+      lookupMethod(classJvmName, methodSignature)!!.parameters
+    }
+    return try {
+      parameters[index]
+          .declaredAnnotations
```

Comment:
Does `Metadata` contain use-site targets info? If yes we should honor them.

Reply:
It does, but afaik only for declared ones. For implicit ones, it's not specified afaik